transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/LegacyFlgDestination.java

```diff
@@ -46,22 +61,40 @@ public String getName() {
   @NonNull
   @Override
   public byte[] getExtras() {
-    return encodeString(apiKey);
+    return encodeAsExtras();
   }
 
   @NonNull
-  static byte[] encodeString(@NonNull String s) {
+  public static LegacyFlgDestination parseFromExtras(@NonNull byte[] a) {
     try {
-      return s.getBytes("UTF-8");
+      String encodedExtra = new String(a, "UTF-8");
+      if (!encodedExtra.startsWith(EXTRAS_VERSION_MARKER)) {
+        throw new IllegalStateException("Version marker missing from extras");
+      }
+      encodedExtra = encodedExtra.substring(EXTRAS_VERSION_MARKER.length());
+      @SuppressWarnings("StringSplitter")
+      String[] fields = encodedExtra.split(EXTRAS_DELIMITER);
```

Comment:
Instead of suppressing the warning pls use 
```suggestion
      String[] fields = encodedExtra.split(EXTRAS_DELIMITER, 2);
```